### PR TITLE
Create a solutions page for Mattermost Customizable ChatGPT Bot Framework

### PR DESCRIPTION
**Pull request by jasonblais**
Hey team!

We're creating a documentation page before we release a cool announcement around AI next week. We wanna merge this PR by the end of the week, so we'd appreciate some feedback from you @azigler, @cwarnermm, and @crspeller.

@azigler, can you take a look at it from a user's perspective? Let me know if there's enough info to get started and if we sound too "marketing-y".

@cwarnermm, would love your editing skills to catch any typos or errors.

And @crspeller, can you give it a "sniff test" and make sure it's up to our standards (and doesn't make false claims)?

Thanks for your help!

### CODE REVIEW

File: source/about/mattermost-customizable-chatgpt-bot-framework.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+The customizable ChatGPT bot framework allows users to integrate generative AI solutions like OpenAI's ChatGPT and private cloud LLMs (Large Language Models) into their Mattermost workflows. This integration can streamline communication, enhance decision-making, and automate tasks, enabling increased speed and efficiency in government, defense, and technology organizations.
+
+An example of the customizable ChatGPT bot framework is the `open source ChatGPT plugin for Mattermost <https://github.com/Brightscout/mattermost-plugin-openai>`_ developed by Brightscout. To get started, follow these steps:
+
+1. Visit the `GitHub repository <https://github.com/Brightscout/mattermost-plugin-openai>`_ for the ChatGPT plugin.
+2. Download the `latest release from the Releases <https://github.com/Brightscout/mattermost-plugin-openai/releases>`_ section.
+3. Upload the release file to your Mattermost server via **System Console > Plugins > Plugin Management**.
+4. Enable the plugin and configure it using your OpenAI API key.
+5. Create a bot account in Mattermost and copy the bot access token.
```
Comment (azigler) [inline source/about/mattermost-customizable-chatgpt-bot-framework.rst:26]: @jasonblais I don't believe steps 5 and 6 are necessary for this plugin. They're not listed anywhere on the repo. I think these steps were only needed for Sebastian Muller's plugin.

File: source/about/mattermost-customizable-chatgpt-bot-framework.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+6. Input the bot access token in the plugin's settings.
+
+Now, you can start using the ChatGPT plugin in your Mattermost instance.
+
+Two other examples of ChatGPT integrations built by Sebastian Müller include:
+
+- `ChatGPT Mattermost Bot with OpenAI <https://github.com/yGuy/chatgpt-mattermost-bot>`_
+- `ChatGPT Mattermost Bot with LLaMa <https://github.com/yGuy/chatgpt-mattermost-bot/tree/llama>`_
```
Comment (azigler) [inline source/about/mattermost-customizable-chatgpt-bot-framework.rst:34]: I recommend adding a `*` at the end of this line and the beginning of the LLaMa disclaimer, to enforce the connection.

File: source/about/mattermost-customizable-chatgpt-bot-framework.rst [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+Mattermost Customizable ChatGPT Bot Framework
+========================================================
+
+Mattermost is committed to providing an open source platform for secure collaboration and control, especially in strict security environments. With a focus on AI augmentation and integration, Mattermost is providing an open source customizable ChatGPT bot framework with full control and data privacy for organizations that want to deploy AI solutions.
+
+Control and Data Privacy
+--------------------------------
+
+Mattermost's open source platform provides verifiable privacy assurances, data control, and granular permissions solutions. As AI becomes more prevalent globally, organizations need control over the AI's access to their data and the ability to use private cloud AI solutions like Azure AI or AWS.
```
Comment (azigler) [inline source/about/mattermost-customizable-chatgpt-bot-framework.rst:11]: Adding links will help open the door for brainstorming and can help them envision their AI on a Mattermost deployment.

Azure AI: https://azure.microsoft.com/en-us/solutions/ai/#benefits
AWS AI: https://aws.amazon.com/machine-learning/ai-services/

File: source/about/mattermost-customizable-chatgpt-bot-framework.rst [hunk 4]
```diff
@@ -0,0 +1,41 @@
+.. _mattermost-customizable-chatgpt-bot-framework:
+
+Mattermost Customizable ChatGPT Bot Framework
+========================================================
+
+Mattermost is committed to providing an open source platform for secure collaboration and control, especially in strict security environments. With a focus on AI augmentation and integration, Mattermost is providing an open source customizable ChatGPT bot framework with full control and data privacy for organizations that want to deploy AI solutions.
```
Comment (azigler) [inline source/about/mattermost-customizable-chatgpt-bot-framework.rst:6]: Proposing:

"Mattermost is providing an open source customizable ChatGPT bot framework with full control and data privacy for organizations that want to deploy AI solutions **without sacrificing security**."

I think driving the security point again at the end strengthens the proposed copy.

**Review by azigler — APPROVED** [review]
@jasonblais From a user perspective, I think everything I would want to read is provided here. There isn't much to dive into yet, but this PR does a good job of laying out options. I added some constructive feedback inline where it made sense to me.

Separately, I'm wondering if you have thoughts on adding AskSage as an industry-specialized example. I think it's too premature to add yet (which makes sense for why it's not here) but do you think it would eventually make an appearance on this page?

The Contact Sales CTA at the end is the only thing that really comes off "too market-y" to me, but I think we should start with it and see what kind of engagement/form abandonment we get from that link. It would be good data to keep an eye on. cc @nat-gunner

File: source/about/mattermost-customizable-chatgpt-bot-framework.rst [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+The customizable ChatGPT bot framework allows users to integrate generative AI solutions like OpenAI's ChatGPT and private cloud LLMs (Large Language Models) into their Mattermost workflows. This integration can streamline communication, enhance decision-making, and automate tasks, enabling increased speed and efficiency in government, defense, and technology organizations.
+
+An example of the customizable ChatGPT bot framework is the `open source ChatGPT plugin for Mattermost <https://github.com/Brightscout/mattermost-plugin-openai>`_ developed by Brightscout. To get started, follow these steps:
+
+1. Visit the `GitHub repository <https://github.com/Brightscout/mattermost-plugin-openai>`_ for the ChatGPT plugin.
+2. Download the `latest release from the Releases <https://github.com/Brightscout/mattermost-plugin-openai/releases>`_ section.
+3. Upload the release file to your Mattermost server via **System Console > Plugins > Plugin Management**.
+4. Enable the plugin and configure it using your OpenAI API key.
+
+Now, you can start using the ChatGPT plugin in your Mattermost instance.
```
Comment (crspeller) [inline source/about/mattermost-customizable-chatgpt-bot-framework.rst:29]:
```suggestion
Now, you can start using the OpenAI plugin in your Mattermost instance.
```

File: source/about/mattermost-customizable-chatgpt-bot-framework.rst [hunk 6]
```diff
@@ -0,0 +1,41 @@
+.. _mattermost-customizable-chatgpt-bot-framework:
+
+Mattermost Customizable ChatGPT Bot Framework
```
[the remark this answers is not in the document]
Comment (azigler) [reply source/about/mattermost-customizable-chatgpt-bot-framework.rst:3]: Good clarification.

[the remark this answers is not in the document]
Comment (jasonblais) [reply source/about/mattermost-customizable-chatgpt-bot-framework.rst:3]: @crspeller Thanks for the feedback!

[On our other thread](https://docs.google.com/document/d/1Y9Z3kfzwP07IsqO8VjAAmetiaMg6flGrFLcVgc3EgOc/edit?disco=AAAAvRhIfbQ) we talked a doc page for "Mattermost customizable ChatGPT bot framework" with a vision of generative AI bot integrations, including links to examples and tradeoffs (e.g. one being unavailable for commercial use). 

Would there be a way to adjust the title and/or introduction to this plan?

File: source/about/mattermost-customizable-chatgpt-bot-framework.rst [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+========================================================
+
+Mattermost is committed to providing an open source platform for secure collaboration and control, especially in strict security environments. 
+
+With a focus on AI augmentation and integration, Mattermost is providing an open source customizable ChatGPT bot framework with full control and data privacy for organizations that want to deploy AI solutions without sacrificing security.
+
+Control and Data Privacy
+--------------------------------
+
+Mattermost's open source platform provides verifiable privacy assurances, data control, and granular permissions solutions. As AI becomes more prevalent globally, organizations need control over the AI's access to their data and the ability to use private cloud AI solutions like `Azure AI <https://azure.microsoft.com/en-us/solutions/ai/#benefits>`_ or `AWS <https://aws.amazon.com/machine-learning/ai-services/>`_.
```
Comment (cwarnermm) [inline source/about/mattermost-customizable-chatgpt-bot-framework.rst:13]:
```suggestion
Mattermost's open source platform provides verifiable privacy assurances, data control, and granular permissions solutions. As AI becomes more prevalent globally, organizations need control over the AI's access to their data, and the ability to use private cloud AI solutions like `Azure AI <https://azure.microsoft.com/en-us/solutions/ai/#benefits>`__ or `AWS <https://aws.amazon.com/machine-learning/ai-services/>`__.
```

File: source/about/mattermost-customizable-chatgpt-bot-framework.rst (hunk 6 above)
Comment (cwarnermm) [inline source/about/mattermost-customizable-chatgpt-bot-framework.rst:3]:
```suggestion
Mattermost customizable ChatGPT bot framework
```

File: source/about/mattermost-customizable-chatgpt-bot-framework.rst [hunk 8]
```diff
@@ -0,0 +1,41 @@
+.. _mattermost-customizable-chatgpt-bot-framework:
+
+Mattermost Customizable ChatGPT Bot Framework
+========================================================
+
+Mattermost is committed to providing an open source platform for secure collaboration and control, especially in strict security environments. 
+
+With a focus on AI augmentation and integration, Mattermost is providing an open source customizable ChatGPT bot framework with full control and data privacy for organizations that want to deploy AI solutions without sacrificing security.
+
+Control and Data Privacy
```
Comment (cwarnermm) [inline source/about/mattermost-customizable-chatgpt-bot-framework.rst:10]:
```suggestion
Control and data privacy
```

File: source/about/mattermost-customizable-chatgpt-bot-framework.rst [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+--------------------------------
+
+Mattermost's open source platform provides verifiable privacy assurances, data control, and granular permissions solutions. As AI becomes more prevalent globally, organizations need control over the AI's access to their data and the ability to use private cloud AI solutions like `Azure AI <https://azure.microsoft.com/en-us/solutions/ai/#benefits>`_ or `AWS <https://aws.amazon.com/machine-learning/ai-services/>`_.
+
+In the short term, this may involve integrating private-cloud AI with Mattermost, while in the long term, it could include training foundation models with an organization's data in a controlled environment.
+
+Get Started with the Sample Open Source ChatGPT Plugin
```
Comment (cwarnermm) [inline source/about/mattermost-customizable-chatgpt-bot-framework.rst:17]:
```suggestion
Get started with the sample open source ChatGPT plugin
```

File: source/about/mattermost-customizable-chatgpt-bot-framework.rst [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+--------------------------------------------------------
+
+The customizable ChatGPT bot framework allows users to integrate generative AI solutions like OpenAI's ChatGPT and private cloud LLMs (Large Language Models) into their Mattermost workflows. This integration can streamline communication, enhance decision-making, and automate tasks, enabling increased speed and efficiency in government, defense, and technology organizations.
```
Comment (cwarnermm) [inline source/about/mattermost-customizable-chatgpt-bot-framework.rst:20]:
```suggestion
The customizable ChatGPT bot framework allows users to integrate generative AI solutions like OpenAI's ChatGPT and private cloud LLMs (Large Language Models) into their Mattermost workflows. This integration can streamline communication, enhance decision-making, and automate tasks to enable increased speed and efficiency in government, defense, and technology organizations.
```

File: source/about/mattermost-customizable-chatgpt-bot-framework.rst [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+An example of the customizable ChatGPT bot framework is the `open source ChatGPT plugin for Mattermost <https://github.com/Brightscout/mattermost-plugin-openai>`_ developed by Brightscout. To get started, follow these steps:
+
+1. Visit the `GitHub repository <https://github.com/Brightscout/mattermost-plugin-openai>`_ for the ChatGPT plugin.
+2. Download the `latest release from the Releases <https://github.com/Brightscout/mattermost-plugin-openai/releases>`_ section.
```
Comment (cwarnermm) [inline source/about/mattermost-customizable-chatgpt-bot-framework.rst:25]:
```suggestion
2. Download the `latest release <https://github.com/Brightscout/mattermost-plugin-openai/releases>`__ from the **Releases** section.
```

File: source/about/mattermost-customizable-chatgpt-bot-framework.rst [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+3. Upload the release file to your Mattermost server via **System Console > Plugins > Plugin Management**.
```
Comment (cwarnermm) [inline source/about/mattermost-customizable-chatgpt-bot-framework.rst:26]:
```suggestion
3. Upload the release file to your Mattermost server by going to **System Console > Plugins > Plugin Management**.
```

File: source/about/mattermost-customizable-chatgpt-bot-framework.rst [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+4. Enable the plugin and configure it using your OpenAI API key.
+
+Now, you can start using the ChatGPT plugin in your Mattermost instance.
+
+Two other examples of ChatGPT integrations built by Sebastian Müller include:
+
+- `ChatGPT Mattermost Bot with OpenAI <https://github.com/yGuy/chatgpt-mattermost-bot>`_
```
Comment (cwarnermm) [inline source/about/mattermost-customizable-chatgpt-bot-framework.rst:33]:
```suggestion
- `ChatGPT Mattermost Bot with OpenAI <https://github.com/yGuy/chatgpt-mattermost-bot>`__
```

**Review by cwarnermm — COMMENTED** [review]
Great draft, @jasonblais! I've provided editorial feedback inline for consideration.

File: source/about/mattermost-customizable-chatgpt-bot-framework.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (crspeller) [reply source/about/mattermost-customizable-chatgpt-bot-framework.rst:3]: Speak in future tense or change the title. It sounds in some places we are describing something that exists. If this is aspirational it should be clear what we have today (the plugins) and what we might build (the framework).

**Comment by jasonblais** [comment]
@crspeller @azigler @cwarnermm Thank you for the feedback, I've made changes to address each.

@crspeller: I've changed the phrasing to be more future-looking.

Let me know if further suggestions before merging.

**Review by crspeller — APPROVED** [review]
Great!

**Comment by azigler** [comment]
@jasonblais LGTM! 👍

File: source/about/mattermost-customizable-chatgpt-bot-framework.rst (hunk 1 above)
Comment (cwarnermm) [reply source/about/mattermost-customizable-chatgpt-bot-framework.rst]: @jasonblais - Please note @azigler's feedback here.

File: source/about/mattermost-customizable-chatgpt-bot-framework.rst (hunk 2 above)
Comment (cwarnermm) [reply source/about/mattermost-customizable-chatgpt-bot-framework.rst]: @jasonblais - I agree with @azigler's feedback here.

**Review by cwarnermm — APPROVED** [review]

**Comment by jasonblais** [comment]
Thanks everyone!

**Pull request merged**